api_gateway/Dockerfile

```diff
@@ -1,8 +1,11 @@
 FROM python:3.8
 
-COPY requirements.txt /
+COPY api_gateway/requirements.txt /
 RUN pip install -r requirements.txt
 
+COPY api_gateway /opt/api_gateway
```

Comment:
why do we need this in build time?

Reply:
Without it, we would just have an empty python image. Even if the code is mounted as a volumen to speed up development (not having to rebuild to try changes), the code should still be in the image so we can use it once we want to deploy with kubernetes or just docker.